packages/plugin-react/src/index.ts

```diff
@@ -0,0 +1,258 @@
+          // commonly used with TS.
+          parserPlugins.push('typescript', 'decorators-legacy')
+        }
+
+        const isNodeModules = id.includes('node_modules')
+        if (!isNodeModules && !isProduction) {
+          plugins.push(
+            await interopDefault(
+              import('@babel/plugin-transform-react-jsx-self')
+            ),
+            await interopDefault(
+              import('@babel/plugin-transform-react-jsx-source')
+            )
+          )
+        }
+
+        let useFastRefresh = false
+        if (!skipFastRefresh && !ssr && !isNodeModules) {
+          // Modules with .js or .ts extension must import React.
+          const isReactModule = id.endsWith('x') || code.includes('react')
+          if (isReactModule && filter(id)) {
+            useFastRefresh = true
+            plugins.push([
+              await import('react-refresh/babel'),
+              { skipEnvCheck: true }
+            ])
+          }
+        }
+
+        let ast: t.File | null | undefined
+        if (id.endsWith('x')) {
+          if (opts.jsxRuntime === 'automatic') {
+            // By reverse-compiling "React.createElement" calls into JSX,
+            // React elements provided by dependencies will also use the
+            // automatic runtime!
+            const [restoredAst, isCommonJS] = isNodeModules
+              ? await restoreJSX(babel, code)
+              : [null, false]
+
+            ast = restoredAst
+
+            plugins.push([
+              await import('@babel/plugin-transform-react-jsx'),
+              { runtime: 'automatic' }
+            ])
+
+            // Avoid inserting `import` statements into CJS modules.
+            if (isCommonJS) {
+              plugins.push(babelImportToRequire)
+            }
+          }
+          // Even if the automatic JSX runtime is not used, we can still
+          // inject the React import for .jsx and .tsx modules.
+          else if (!isNodeModules && !/(^|\n)import React /.test(code)) {
+            code = `import React from 'react'; ` + code
+          }
+        }
+
+        const isReasonReact = id.endsWith('.bs.js')
+
+        const babelOpts: TransformOptions = {
+          babelrc: false,
+          configFile: false,
+          ...opts.babel,
+          ast: !isReasonReact,
+          root: projectRoot,
+          filename: id,
+          parserOpts: {
+            ...opts.babel?.parserOpts,
+            sourceType: 'module',
+            allowAwaitOutsideFunction: true,
+            plugins: parserPlugins
+          },
+          generatorOpts: {
+            ...opts.babel?.generatorOpts,
+            decoratorsBeforeExport: true
+          },
+          plugins,
+          sourceMaps: true,
+          sourceFileName: id
+        }
+
+        const result = ast
+          ? await babel.transformFromAstAsync(ast, code, babelOpts)
+          : await babel.transformAsync(code, babelOpts)
```

Comment:
Instead of importing from babel, seems the compiler doesn't like it and prefers to import the function directly (otherwise complains that babel object doesn't have these functions/imports.

```suggestion
          ? await transformFromAstAsync(ast, code, babelOpts)
          : await transformAsync(code, babelOpts)
```

And in the top extend the babel import to 

```
import babel, { ParserOptions, TransformOptions, types as t, transformAsync, transformFromAstAsync } from '@b
abel/core'
```